### DISCUSSION TITLE
How do you use @sentry/webpack-plugin with rsbuild?

### DISCUSSION THREAD

**Discussion by boosja** [web-infra-dev/rsbuild#3950] [Q&A] [answered]
[Here](https://rspack.dev/guide/compatibility/plugin) it says the plugin should be compatible, but when I add it to tools.rspack and run `rsbuild dev`, I get the error:
```
error   Failed to load file with jiti: ***/rsbuild.config.ts
error   Failed to start dev server.
error   Cannot find module 'webpack'
```

This is how I add the plugin to `rsbuild.config.ts`
```
tools: {
  rspack: (_config, { appendPlugins }) => {
    appendPlugins([
      sentryWebpackPlugin({
        // config...
      }),
    ])
  },
},
```

Is it something I did wrong or is the plugin not longer compatible with rsbuild?

@sentry/webpack-plugin: `2.22.6`
@rsbuild/core: `1.0.1-rc.5`
@rsbuild/plugin-react: `1.0.1-rc.5`

**Comment by 9aoy** [comment, marked as answer]
`webpack` is a peer dep of `@sentry/webpack-plugin`, so you need install `webpack` in your project, although it is not actually used in rsbuild.

**Reply by boosja** [reply]
Ah, that's why. Thanks for the pointer! 😄

**Comment by chenjiahan** [comment]
I just checked the source code of `@sentry/webpack-plugin` and it still needs to be compatible with webpack 4. That's why it requires a `webpack` dependency to be installed, actually the `webpack` package is never used in Rspack or Rsbuild projects.

@9aoy The `@sentry/webpack-plugin` package is quite simple, maybe we can submit a PR to make `webpack` an optional dependency.
